src/coq/decompiler/decompiler.ml

```diff
@@ -63,14 +92,25 @@ let run_tac env sigma (tac : unit Proofview.tactic) (goal : constr)
     
 (* Returns true if the given tactic solves the goal. *)
 let solves env sigma (tac : unit Proofview.tactic) (goal : constr) =
-  fst (run_tac env sigma tac goal) = []
+  let subgoals, _ = run_tac env sigma tac goal in subgoals = []
```

Comment:
What is the second argument here that is dropped? Is it an `evar_map`? If so, you might need to return it (sorry I missed this earlier)

Reply:
I think this also needs error handling: if this results in an error, return false?